sale_commission_geo_assign/models/res_partner.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+from odoo import models, fields, api
+
+
+class Partner(models.Model):
+    _inherit = 'res.partner'
+
+    country_ids = fields.Many2many(
+        'res.country', string='Countries',
+        help='Countries where this agent operates'
+    )
+    state_ids = fields.Many2many(
+        'res.country.state', string='States',
+        help='States where this agent operates'
+    )
+    zip_from = fields.Char(
+        'Zip From', help='ZIP range where this agent operates')
+    zip_to = fields.Char(
+        'Zip To', help='ZIP range where this agent operates')
+
+    @api.onchange('country_ids')
+    def onchange_countries(self):
+        if self.country_ids:
+            domain = [('country_id', 'in', self.country_ids.ids)]
+        else:
+            domain = []
+        return {'domain': {
+            'state_ids': domain
+        }}
+
+    @api.multi
+    def verify_agent(self, partner):
```

Comment:
Please add a little documentation to make it clear what this method does: `Check if it is ok to assign this agent to the partner` or similar